homeassistant/components/stream/worker.py

```diff
@@ -11,104 +11,190 @@
     MAX_MISSING_DTS,
     MAX_TIMESTAMP_GAP,
     MIN_SEGMENT_DURATION,
+    MIN_SEGMENT_DURATION_LL_HLS,
     PACKETS_TO_WAIT_FOR_AUDIO,
+    PART_DURATION,
     SEGMENT_CONTAINER_FORMAT,
     STREAM_TIMEOUT,
 )
-from .core import Segment, StreamBuffer
+from .core import Part, Segment
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def create_stream_buffer(video_stream, audio_stream, sequence):
-    """Create a new StreamBuffer."""
-
-    segment = io.BytesIO()
-    container_options = {
-        # Removed skip_sidx - see https://github.com/home-assistant/core/pull/39970
-        "movflags": "frag_custom+empty_moov+default_base_moof+frag_discont+negative_cts_offsets",
-        "avoid_negative_ts": "disabled",
-        "fragment_index": str(sequence),
-    }
-    output = av.open(
-        segment,
-        mode="w",
-        format=SEGMENT_CONTAINER_FORMAT,
-        container_options={
-            "video_track_timescale": str(int(1 / video_stream.time_base)),
-            **container_options,
-        },
-    )
-    vstream = output.add_stream(template=video_stream)
-    # Check if audio is requested
-    astream = None
-    if audio_stream and audio_stream.name in AUDIO_CODECS:
-        astream = output.add_stream(template=audio_stream)
-    return StreamBuffer(segment, output, vstream, astream)
-
-
 class SegmentBuffer:
     """Buffer for writing a sequence of packets to the output as a segment."""
 
-    def __init__(self, outputs_callback) -> None:
+    def __init__(self, outputs_callback, ll_hls) -> None:
         """Initialize SegmentBuffer."""
         self._stream_id = 0
-        self._video_stream = None
-        self._audio_stream = None
         self._outputs_callback = outputs_callback
         # Each element is a StreamOutput
         self._outputs = []
-        self._sequence = 0
+        self._sequence = -1
```

Comment:
Since we call `reset()` which increments the `_sequence` before we start, we can initialize this to -1 in order to start from 0. Starting at 0 is more consistent with our part indexing as well as HLS default indexing.